src/utils/components/FloatingPosition.jsx

```diff
@@ -36,24 +37,46 @@ class FloatingPosition extends React.PureComponent {
 		const positionTarget = getTrigger().offsetParent ? getTrigger().offsetParent : getTrigger();
 		const positionData = triggerClientRect || positionTarget.getBoundingClientRect();
 		const contentHeight = getContent && getContent().getBoundingClientRect().height;
-
+		const contentWidth = getContent && getContent().getBoundingClientRect().width;
+		const scrollTop = window.scrollY || window.pageYOffset;
+		const scrollLeft = window.scrollX || window.pageXOffset;
 		const {
 			left,
 			top,
 			width,
 			height
 		} = positionData;
-
-		const scrollTop = window.scrollY || window.pageYOffset;
-		const scrollLeft = window.scrollX || window.pageXOffset;
+		const getAdjustedAlignment = alignment => {
```

Comment:
You could define this as an exported pure function outside of the render method by passing in what the function needs instead of using closure scope to access those values.

`export const getAdjustedAlignment = (preferredAlignment, positionData, viewportWidth) => {...}`

and call it like so in the component...
`align: getAdjustedAlignment(alignment, positionData, window.innerWidth)`

and in tests...
````
const expected = 'right';
const actual = getAdjustedAlignment('left', MOCK_POSITION_DATA, MOCK_VIEWPORT_WIDTH);

expect(actual).toBe(expected);
````